_playwright-tests/tests/API/RepositoryFilteringTest.spec.ts

```diff
@@ -0,0 +1,264 @@
+import { expect, test } from 'test-utils';
+import { RepositoriesApi, ApiRepositoryResponse, GetRepositoryRequest } from 'test-utils/client';
+import { poll, randomName, cleanupRepositories } from 'test-utils/helpers';
+
+/**
+ * Tests that repository listings can be filtered by various parameters including
+ * architecture, version, UUID, URL, text search and status.
+ */
+
+test.describe('Repository Filtering Test', () => {
+  const createdRepos: ApiRepositoryResponse[] = [];
+
+  test('should filter repository listings with various parameters including status', async ({
+    client,
+    cleanup,
+  }) => {
+    const baseRepoName = `Filter-Test-${randomName()}`;
+    const repositoriesApi = new RepositoriesApi(client);
```

Comment:
this is used only in a few steps, for consistency i'd suggest either using it everywhere or removing it